src/recorder.js

```diff
@@ -228,6 +247,10 @@ Recorder.prototype.finish = function() {
     this.ondataavailable( outputData );
   }
   this.onstop();
+  if ( this._finished ) {
+    this._finished();
```

Comment:
Same here. Whats the need for a promise in addition to the `onstop` callback? Is it just for testing?